<launch>
    <node name="camera0"
        pkg="easyscan"
        type="camera_node"
        output="screen"
        >
        <remap from="camera/image_raw" to="camera0/image_raw"/>
 
        <param name="numberCameraName" type="bool" value="false"/>
        <param name="cameraName" type="string" value="nvcamerasrc ! video/x-raw(memory:NVMM), width=(int)640, height=(int)480,format=(string)I420, framerate=(fraction)24/1 ! nvvidconv flip-method=0 ! video/x-raw, format=(string)BGRx ! videoconvert ! video/x-raw, format=(string)BGR ! appsink"/> 
        <param name="needCameraInfo" type="bool" value="true"/>
        <param name="cameraCalibrationFileName" type="string" value="$(find easyscan)/config/ost0.yaml"/>

    </node>
 
    <node name="camera1"
        pkg="easyscan"
        type="camera_node"
        output="screen"
    >
        <remap from="camera/image_raw" to="camera1/image_raw"/>
        <param name="numberCameraName" type="bool" value="true"/>
        <param name="cameraName" type="int" value="1"/>
        <param name="needCameraInfo" type="bool" value="true"/>
        <param name="cameraCalibrationFileName" type="string" value="$(find easyscan)/config/ost1.yaml"/>
 
    </node>

    <node name="localize_node" pkg="easyscan" type="localize_QRcode.py" output="screen">
        <param name="imageNode" type="string" value="camera0/image_rect_color"/>
        <param name="infoNode" type="string" value="camera0/camera_info"/>
    </node>

    <node name="laser_node" pkg="easyscan" type="laser_process.py" output="screen">
        <param name="imageNode" type="string" value="camera1/image_rect_color"/>
    </node>

    <node name="image_proc0" pkg="image_proc" type="image_proc" ns="camera0"> </node>
    <node name="image_proc1" pkg="image_proc" type="image_proc" ns="camera1"> </node>
 
    <node name="calibrate_camera1_QRcode"
        pkg="easyscan"
        type="calibrate_camera1_QRcode.py"
        output="screen">

        <param name="image0Node" type="string" value="camera0/image_rect_color"/>
        <param name="image1Node" type="string" value="camera1/image_rect_color"/>
        <param name="size" type="string" value="8x3"/>
        <param name="maxImageNum" type="int" value="10"/>
        <param name="square" type="double" value="0.02"/>
        
    </node> 
    
</launch>
